sklearn/tests/test_discriminant_analysis.py

```diff
@@ -176,8 +175,12 @@ def test_lda_explained_variance_ratio():
     clf_lda_svd = LinearDiscriminantAnalysis(solver="svd")
     clf_lda_svd.fit(X, y)
     assert_almost_equal(clf_lda_svd.explained_variance_ratio_.sum(), 1.0, 3)
+
+    # NOTE: clf_lda_eigen.explained_variance_ratio_ is not of n_components
+    # length. Make it the same length as clf_lda_svd.explained_variance_ratio_
+    # before comparison.
     assert_array_almost_equal(clf_lda_svd.explained_variance_ratio_,
-                              clf_lda_eigen.explained_variance_ratio_)
+            clf_lda_eigen.explained_variance_ratio_[:3])
```

Comment:
@JPFrancoia: Sorry, your'e right. They're not of the same length.

I think what would be more transparent than the 3 is using `tested_length = min(clf_lda_svd.explained_variance_ratio_.shape[0], clf_lda_eigen.explained_variance_ratio_.shape[0])`.

The extra length is caused by precision errors in this case, not because of the issue mentioned in #6032. #6032 deals with the fact when `n_components` is set in the constructor, but this length is not reflected by `explained_variance_ratio_`. On the other hand, in this case, we are not setting `n_components` in either constructor, but we're getting different lengths due to precision errors.